docs: Add some guidelines for term editors. (#2010)
obophenotype/cell-ontology: docs/editing_guidelines.md

# Guidelines for CL editors

## Naming terms

### General rules

> 1. All names should be singular nouns.

Reviewer: ```suggestion
1. All labels should be singular nouns.
```

Reviewer 2: I agree that "label" or "labels" should be used throughout rather than "name" or "names", particularly as the annotation property used is 'label'.

> 2. Names should not be capitalised, unless they are or contain a proper name (e.g. _Johnston's organ_).

Reviewer: ```suggestion
2. Names should not be capitalised unless they are or contain a proper name (e.g., _Johnston's organ_).
```

Reviewer 3: This is explicitly a rule about naming.  The change makes is a rule about words.

Reviewer 3: I guess the context helps - but the main aim is to ensure editors avoid the habit of capitalising the first letter of a full name.

> 3. Avoid special characters: use only alpha-numeric characters, space, dash (`-`), slash (`/`), and apostrophe (`'`).

Reviewer: ```suggestion
3. Avoid special characters: use only alphanumeric characters, space, dash (`-`), slash (`/`), and apostrophe (`'`).
```

### Advice

> Bear in mind that users will often encounter terms in isolation. Long descriptive names (within reason!) are therefore preferable, especially where there is obvious potential for confusion. For example, _calyx_ (which simply means “cup”) could refer to a structure in the oviduct or in the mushroom body, depending on your field of specialisation. It is therefore better to use _mushroom body calyx_ and _oviduct calyx_ rather than simply _calyx_ alone.

Reviewer: ```suggestion
Bear in mind that users will often encounter terms in isolation. Long, descriptive labels (within reason!) are therefore preferable, especially where there is obvious potential for confusion. For example, _calyx_ (which simply means “cup”) could refer to a structure in the oviduct or in the mushroom body, depending on your field of specialization. It is therefore better to use _mushroom body calyx_ or _oviduct calyx_ rather than simply _calyx_ alone.
```

## Defining terms
This section is about the _textual_ definition of terms. Logically consistent classification is important, but an ontology is only useful (and maintainable) if all humans that interact with it (users, curators and editors) can quickly find the terms they need and understand what they refer to. This requires clear, unambiguous, human-readable definitions.

When crafting a definition, editors should aim for a reasonably succinct statement about the class allowing curators and users to easily distinguish it from other, similar classes, and which captures key points of interest about that class. It should capture assertions made in the formal part of the definition (the relationships) as closely as possible without becoming stilted and difficult to read.

The basic structure of a definition should be as follows:

> A `genus` that `diff1` and `diff2`. It also `diff3` and `gloss`...

where `genus` is a general classification and `diff1`, `diff2`, etc. are the _differentia_, which state what differentiates this class from others that share the same general classification. The gloss, when present, gives some key points of interest about the class.

> The first sentence of the definition should be in singular form. The remaining of the definition may then enter into plural if needed.

Reviewer: ```suggestion
The first sentence of the definition should refer to the definiendum in singular form. The rest of the definition may then invoke the plural form.
```

### Contents of definitions
It is difficult to specify, _a priori_, which assertions should be included in a textual definition. However there are some general guidelines.

1. DO make sure your definition is consistent with the definition of the superclass(es).
2. DO make sure your definition includes the information that is recorded in all the direct formal relationships to the class.
3. AVOID assertions about structures or cell types that are not part of the cell type being described, except when they pertain to some direct relationships with the cell type being described.
4. AVOID including details that could better be included in the definition of subtypes of the cell type being described.
5. LIMIT information that applies to only some members of the class. This should only be used sparingly; when used, it should be made clear that it does not apply to all members of the class.
6. AVOID using gene expression as the differentium.
7. AVOID extensive repetition of assertions made in superclass definitions, unless these assertions are used to provide direct evidence for class membership.

> 8. DO NOT add informations about what happens in mutant backgrounds or other kind of non-control conditions.

Reviewer: ```suggestion
8. DO NOT add information about what happens in pathological states or other kind of non-control conditions.
```

9. DO NOT include reasons for believing the assertions to be true. These should be recorded in comments.
10. DO NOT raise questions in the definition. The definition should have the sense of being definitive, which is undermined if we show doubt. Any doubt should be recorded in comments instead.

## Comments
Comments should be used for:

1. Providing evidence. In some cases it is useful to know the type of evidence for an assertion. This should not be recorded in the definition, but _can_ be recorded in a comment.

> 2. Disambiguation. Sometimes a single term is used in the literature with multiple meanings. In such cases, acomment should be added outlining these different uses and how they relate to the standard definition set in the ontology.

Reviewer: ```suggestion
2. Disambiguation. Sometimes a single term is used in the literature with multiple meanings. In such cases, a comment should be added outlining these different uses and how they relate to the definition set in the ontology.
```

> 3. Reporting editorial decisions (or decisions in waiting) about the term. This includes providing a reason for obsoleting a term, or letting users and curators know that the term may be merged or split in the future, e.g. when enough evidence for the merge/split will be available.

Reviewer: ```suggestion
3. Reporting editorial decisions (or decisions in waiting) about the term. This includes providing a reason for obsoleting a term, or letting users and curators know that the term may be merged or split in the future, e.g., when enough evidence for the merge/split will be available.
```

Try to be consistent in how you phrase the various types of comments. For example:

* when giving a reason for obsoletion, use “Obsoleted as ...”;
* when indicating a potential future merge, use “Possible equivalence with {other term} ...”.

## Synonyms

> Extensive addition of synonyms help searching. Add as many synonyms as you like.

Reviewer: ```suggestion
Extensive addition of synonyms helps "findability" of terms when searching. Accurate synonyms can and should be added liberally.
```

Guidelines on the type of synonyms:

1. Use an _exact_ synonym when the name and the synonym can be used interchangeably, at least in a given context. This includes the case where the synonym is an abbreviation that, in a broader context, could have a different meaning. For example, _DPM_  can be an exact synonym of _dorsal pharyngeal muscle cell_, even though “DPM” could have many more meanings generally, because in the context where “dorsal pharyngeal muscle cell” is used, “DPM” is commonly used interchangeably to refer to these cells.

> 2. There is no expectation that exact synonyms should be unique across the entire ontology. For example, in the _Drosophila_ Anatomy Ontology _DPM_ is also an exact synonym of _mushroom body dorsal paired medial cell_.

Reviewer: ```suggestion
2. Exact synonyms should be unique across the ontology. In other words, if class _A_ has exact synonym "_X_", "_X_" should not be an exact synonym for any other CL term.
```

Author: This requires further discussion.

For what it’s worth, I disagree.

1. Such a policy, if we indeed want to have it, should be enforced in CI checks, not merely stated in guidelines.
2. This would mean that synonyms that may have several meanings in a broader context would all end up as being _related_ synonyms. That category is already the “catch-all” category for all synonyms that are neither exact, narrow, or related, making it the least meaningful category. Such a policy would dilute its meaning even more.
3. This makes even less sense when you consider that ultimately, CL is not a standalone ontology but a part of a much larger set. Enforcing uniqueness of exact synonyms within CL is pointless if we cannot enforce uniqueness OBO-wide.

Reviewer: This was discussed and reflected [here](https://github.com/obophenotype/uberon/issues/2363#issuecomment-1446928901). 

>This would mean that synonyms that may have several meanings in a broader context would all end up as being related synonyms.

Yes, annotating terms as exact synonyms should be used conservatively and only when they refer exactly to the concept captured in the ontology.

Author: My bad, I must have missed that particular Uberon call.

I still think it’s a bad idea, but if there is already an agreement on the issue then it’s fine by me.

Author: But point 1 above still stands: if it is the policy of Uberon/CL that exact synonyms should be unique across the ontology, then the policy should be enforced in CI checks. You cannot rely on editors 1) being aware of the policy (hell, I am an editor and I was not aware of it!) and 2) taking the time to manually check for uniqueness at edit time.

Author: @dosumis Well, FBbt and CL already _do_ diverge on that issue. We have a lot of duplicated exact synonyms across the ontology, because we allow synonyms to be “exact in a given context”. And for the record I do not plan to change that in FBbt, whatever CL is doing.

Reviewer 3: For the record, I prefer FBbt's approach but am consistently outvoted in general discussion.  In the end we should judge by some clear statements of use cases.  VFB approach is not great for NLP, but I think is fine for search - users can and should be trusted to do the final disambigutaion.

> 3. Be mindful of the “directionality” of the _narrow_ and _broad_ types of synonyms. They qualify the _synonym_, not the original term. For example, saying that _dmd5_ is a narrow synonym of _abdominal dorsal multidendritic neuron_ means that “dmd5“ refers to a narrower concept than _abdominal dorsal multidendritic neuron_, not the other way around.

Reviewer: ```suggestion
3. Be mindful of the “directionality” of the _narrow_ and _broad_ types of synonyms. They qualify the _synonym_, not the original term. For example, saying that "_peripheral blood mononuclear cell_" is a narrow synonym of '_mononuclear cell_' means that "_peripheral blood mononuclear cell_" refers to a narrower concept than '_mononuclear cell_', not the other way around.
```

> 4. The _related_ type may be used for other cases where the overlap between the synonym and the term name may be unclear, but you want people to find the term when they are searching for the synonym.

Reviewer: ```suggestion
4. The _related_ synonym type should be used for cases where the overlap between the synonym and the term label may be unclear, disputable or not true in all scenarios or contexts, but you want the term to be findable when searching. This includes abbreviations, which should be annotated as _related_ synonyms with synonym type 'abbreviation' (see point 1 above). 
If a synonym includes a mix of abbreviations and words, the _related_ type should still be used unless there is enough context within the synonym itself to make it clear that the synonym refers only to the concept being annotated. For example, "_lung TRM CD8-positive, CD103-positive cell_" would be an _exact_ synonym of '_lung resident memory CD8-positive, CD103-positive, alpha-beta T cell_' even though TRM is an abbreviation for "tissue resident T cell". Compare to "_IMB cell_", which should be a _related_ synonym of '_invaginating midget bipolar cell_'.
```

## Considerations on style
The following considerations apply both to all human-readable fields (names, textual definitions, comments, synonyms).

* **US english or British english?** Where there are differences in the accepted spelling between english and US usage, use the US form. British english variants of the labels may be added as synonyms (e.g., a term labelled “epithelial cell of esophagus” may have “epithelial cell of oesophagus” as an exact synonym).

> * **Use of jargon.** The aim of the ontology is to provide useable descriptions and links to the reader. Consequently, try to avoid obscure jargon or pretentious latin/greek, especially where widely understood, plain alternatives exist. Where it will aid searching, such terms may be added to synonyms and/or as asides in the “gloss” part of definitions.

Reviewer: ```suggestion
* **Use of jargon.** The aim of the ontology is to provide useable descriptions and links to the reader. Consequently, try to avoid obscure jargon or pretentious Latin/Greek, especially where widely understood, plain alternatives exist. Where it will aid searching, such terms may be added to synonyms and/or as asides in the “gloss” part of definitions.
```

> * **That or which?** These are so interchangead that there isn’t really a rule anymore. But as a guideline, use “which” after a comma (e.g., “this study, which cost $10,000, was a success”), and “that” when no comma is used (e.g., “the study that cost $10,000 was a success”).

Reviewer: ```suggestion
* **That or which?** These are so interchangeable that there isn’t really a rule anymore. But as a guideline, use “which” after a comma (e.g., “this study, which cost $10,000, was a success”), and “that” when no comma is used (e.g., “the study that cost $10,000 was a success”).
```

> * **Use of hyphens.** Yes to those that help clarity (e.g., “posterior-most”) and those that are regularly used/accepted in the literature. Generally, no hyphens after prefixes such as “sub”, “mid”, “semi”, “hemi”, etc. (e.g., “hemidesmosome” instead of “hemi-desmosome”), unless it helps with clarity (e.g. “multi-innervated”). No hyphens for composed location adjectives (e.g., “posteroanterior”), unless there are more than two compounds (e.g., “ventro-posterolateral”).

Reviewer: ```suggestion
* **Use of hyphens.** Yes to those that help clarity (e.g., “posterior-most”) and those that are regularly used/accepted in the literature. Generally, no hyphens after prefixes such as “sub”, “mid”, “semi”, “hemi”, etc. (e.g., “hemidesmosome” instead of “hemi-desmosome”), unless it helps with clarity (e.g., “multi-innervated”). No hyphens for composed location adjectives (e.g., “posteroanterior”), unless there are more than two compounds (e.g., “ventro-posterolateral”).
```

ORCID identifiers may also be used when the only available source for an assertion is an individual researcher. This should be done sparingly.

## Formal definitions
The formal definition of a class is made up of all the logical axioms about the class (as opposed to the annotation assertion axioms). This includes classification assertions, relationship assertions, equivalence assertions, and disjointness assertions.

> Note: In OWL formalism, both classification and relationship assertions are represented using `SubClassOf` axioms. However in this document, we make a strict distinction between a _classification_ (where a class is a subclass of a _named class_), and a _relationship_ (where a class is a subclass of an _anonymous class expression_).
>
> For example, in the formal definition of CL:0000392 (“crystal cell”):
>
> ```
> Class: 'crystal cell'
>   SubClassOf: 'hemocyte'
>   SubClassOf: 'develops from' some 'procrystal cell'
> ```
>
> The first `SubClassOf` axiom denotes an actual classification, whereas the second denotes a relationship.

### Asserting classification

> In order to keep the ontology maintainable, asserted classifications should be limited where possible. Ideally, all terms would have only a single asserted parentt. Given the presence of suitable logically defined classes (see below) and sufficient relationships for the term you are making, additional classification can be inferred automatically by reasoning.

Reviewer: ```suggestion
In order to keep the ontology maintainable, asserted classifications should be limited where possible. Ideally, all terms would have only a single asserted parent (also known as "superclass"). Given the presence of suitable logically defined classes (see below) and sufficient relationships for the term you are making, additional classification can be inferred automatically by reasoning.
```

> Strictly speaking, logical definitions can be arbitrarily complex. It is recommended, however, to stick as much as possible to “simple” logical definitions following the structure above. More complex definitions are harder to understand, and may involve OWL constructs that are legal but not fully supported by the available reasoners.

Good candidates for logically defined classes include classes whose only differentium is:

* what the cell is part of;
* what it innervates (or, if more precisely known, what its axon(s) or dendrite(s) innervate);
* its function.

Care must be taken not to use logically defined classes too liberally. You should be satisfied that all automatic classifications that result would make sense (or at least be justifiable) to a biologist and would cover most cases of usages with as few edge cases as possible.

If you are unsure whether a characteristic of a cell should be expressed as a relationship or as a differentium in a logically defined class: use a relationship.

> The textual definition of a logically defined class should be a plain english equivalent of the logical definition. However, in some cases it can be useful to add more details in the “gloss” part, when relevant.

Reviewer: ```suggestion
The textual definition of a logically defined class should be a plain English equivalent of the logical definition. However, in some cases it can be useful to add more details in the “gloss” part when relevant.
```

10. DO NOT raise questions in the definition. The definition should have the sense of being definitive, which is undermined if we show doubt. Any doubt should be recorded in comments instead.

## Comments
Comments should be used for:

1. Providing evidence. In some cases it is useful to know the type of evidence for an assertion. This should not be recorded in the definition, but _can_ be recorded in a comment.
2. Disambiguation. Sometimes a single term is used in the literature with multiple meanings. In such cases, a comment should be added outlining these different uses and how they relate to the definition set in the ontology.
3. Reporting editorial decisions (or decisions in waiting) about the term. This includes providing a reason for obsoleting a term, or letting users and curators know that the term may be merged or split in the future, e.g., when enough evidence for the merge/split will be available.

Try to be consistent in how you phrase the various types of comments. For example:

* when giving a reason for obsoletion, use “Obsoleted as ...”;
* when indicating a potential future merge, use “Possible equivalence with {other term} ...”.

## Synonyms

> Extensive addition of synonyms help “findability” of terms when search. Synonyms can and should be added liberally.

Reviewer 4: ```suggestion
Extensive addition of synonyms helps “findability” of terms when search. Synonyms can and should be added liberally.
```

2. Exact synonyms should be unique across the ontology. In other words, if class _A_ has synonym “X”, “X” should not be an exact synonym for any other CL term.
3. Be mindful of the “directionality” of the _narrow_ and _broad_ types of synonyms. They qualify the _synonym_, not the original term. For example, saying that “peripheral blood mononuclear cell” is a narrow synonym of “mononuclear cell” means that “peripheral blood mononuclear cell“ refers to a narrower concept than “mononuclear cell”, not the other way around.
4. The _related_ synonym type should be used for cases where the overlap between the synonym and the term label may be uncler, disputable or not true in all scenarios or contexts, but you want the term to be findable when searching. This includes abbreviations, which should be annotated as _related_ synonyms with synonym type “abbreviation” (see point 1 above).

> 5. If a synonym includes a mix of abbreviations and words, the _related_ type should still be used unless there is enough context within the synonym itself to make it clear that the synonyms refers only to the concept being annotated. For example, “lung TRM CD8-positive, CD103-positive cell” would be an exact synonym of “lung resident memory CD8-positive, CD103-positive, alpha-beta T cell”, even though “TRM” (in this case) is an abbreviation for “tissue resident T cell”. Note that without this context “TRM” should not be considered an exact synonym for “tissue resident T cell” as “TRM” could also mean “treatment-related mortality”, another OBO ontology concept. Compare the previous example to “IMB cell”, which should be a _related_ synonym of “invaginating midget bipolar cell”.

Reviewer: ```suggestion
5. If a synonym includes a mix of abbreviations and words, the _related_ type should still be used unless there is enough context within the synonym itself to make it clear that the synonym refers only to the concept being annotated. For example, “lung TRM CD8-positive, CD103-positive cell” would be an exact synonym of “lung resident memory CD8-positive, CD103-positive, alpha-beta T cell”, even though “TRM” (in this case) is an abbreviation for “tissue resident T cell”. Note that without this context “TRM” should not be considered an exact synonym for “tissue resident T cell” as “TRM” could also mean “treatment-related mortality”, another OBO ontology concept. Compare the previous example to “IMB cell”, which should be a _related_ synonym of “invaginating midget bipolar cell”.
```

Minor grammar edit.
